### ISSUE TITLE
Push command successful but no changes reflected on Remote Repository file/s.

### ISSUE THREAD

**Issue by prashantakerkar** [git-lfs/git-lfs#1114]
## GIT Push Command in SourceTree

git -c diff.mnemonicprefix=false -c core.quotepath=false push -v --tags --set-upstream origin Import:Import

Branch Import set up to track remote branch Import from origin.

 = [up to date]      Import -> Import
updating local tracking ref 'refs/remotes/origin/Import'
Everything up-to-date

Completed successfully.

**Push command successful but no changes reflected on Remote Repository file/s.**
What could be the Issue ?.

Thanks & Regards,
Prashant S Akerkar

**Comment by sinbad** [comment]
`git push` has concluded that there's nothing to push there, how are you judging that there are missing changes? Do you mean that the LFS content is missing, or that commits are missing in the repo?

**Comment by prashantakerkar** [comment]
The Remote Repository file is not showing the updated changes when opened the file in Notepad after running the Push command.

**Comment by sinbad** [comment]
Sorry, it's still not clear to me what you mean by "Remote Repository file". What does that mean? Do you mean:

1. You're opening the file from the website of your host like GitHub/Bitbucket?
1. You've pulled the changes down to a different client?

Please can you be a lot more specific about *exactly* what you're doing.

**Comment by prashantakerkar** [comment]
Steps
-------

1 I did the changes in the Local Repository File which i commit with teh commit command by adding comments.
2 I run the Push command to update the changes in Remote Repository Server which i have configured in SourceTree.
3 Push command gives success message, but when i open the Remote Server File physically in notepad, the changes are not reflected in the Remote Server File.

**Comment by sinbad** [comment]
So step 3  you've said exactly the same thing. How are doing this: "when i open the Remote Server File in notepad"? What file are you opening, exactly? The remote git repo and lfs store does not store your files in the same way as your local machine does, at all, so saying you opened the "Remote Server File" doesn't make sense.

**Comment by prashantakerkar** [comment]
I have Fetched all the files from the Remote Repository Server using the Fetch / Pull command in my Local Repository which is successful.I have configured the Remote Server Repository in Source Tree. The Remote Server is running on Windows Server 2012 Os. 
I have then made changes in a single .aspx file locally which shows changes made in the Local Git Repository opened in SourceTree. After that i run the Push command so that i can push these files on the Server machine where Remote Repository Files are present. The Push command shows success. I connect to the Remote Server using FTP and download the file which i had modified locally and pushed to the Remote Server Folder. The File downloaded locally from the Remote Server do not show any changes which i had done locally. i.e. The File changes done are not reflected when opened in Notepad.

Thanks & Regards,
Prashant S Akerkar

**Comment by sinbad** [comment]
When you push a file to a remote git repository, it becomes part of the git repo there and you can't possibly download it directly using FTP. You must be downloading something else entirely which I expect is why it appears to you to be unmodified. In the git repo it'll be stored in an object file as something like `objects/fe/1525b4fe84318e881f5bc1df2aa53df4d826e4`, it won't look like the file you uploaded. So if you're expecting some folder on your server to contain an `.aspx` file which matches what you pushed, you're misunderstanding how git works. 

Is this related to LFS in any way? I doubt you'd be storing an .aspx file in LFS.

**Comment by rama-chetu** [comment]
try git push origin HEAD:master

**Comment by rirathod** [comment]
Your git local repo must not be pointing to correct remote repo.

**Comment by AdeyinkaAdegbenro** [comment]
try git diff, if you see the new changes you made, then do  git commit -m 'commit message'; if it returns something like index.lock, file exists, then do cd git and rm index.lock.  or remove it manually via your OS GUI. then commit again, then push. Should work, worked for me.

**Comment by aboelmor** [comment]
Hi prashantakerkar, I understand your concern and actually I am having the same problem; any luck finding a solution?
thank you

**Comment by switheyw** [comment]
I assume that you've tried a 'git reset master --hard' or 'git reset --hard' ?  The first time I push
from my local repository the remote repository is empty.   After the reset, files are present.  The remote git file is created as git init, not git init --bare.

**Comment by aboelmor** [comment]
Thank you switheyw for the reply and sorry I couldn't see it earlier (we had a holiday).
I tried what you said (i added a new git non bare repository and redo the same steps) but i still have the same problem. 
What is ironic though is that when i do, from the remote site, a git stash; The changes are there.

Any idea?

**Comment by kboygit** [comment]
Is GitHub not updating contributions lately for the past 3 days. I noticed that the contributions is not updating , it's not that I wanted the contributions to update just for the the sake of contribution just my .02

**Comment by technoweenie** [comment]
@kboygit: See https://help.github.com/articles/why-are-my-contributions-not-showing-up-on-my-profile/ for our docs on Contributions. Please contact [GitHub Support](https://github.com/contact) if you have further questions.

**Comment by dr4dr4** [comment]
I think I have the same query about pushing to a remote as well and would like it clarified.
This may be one of those occasions where I find out myself just after having posted this but anyway.
I have a local working repo with two branches master & dev. I have setup a remote bare repo and added it as a remote to my local as lib.

First push was to lib from master, worked as expected.
Make a change to a file, stage it, commit it. Push again and it updates file correctly on remote.
----------------------------------------------------------------------------------------------------------------------
checkout dev branch, make a change there, stage it, commit it, git push lib dev.->reports ok
So if i diff these confirm that push is correct, I have two remote branches lib/master & lib/dev.
Physically on the remote the file system only shows the file changes when the master is pushed and not the dev branch.
-----------------------------------------------------------------------------------------------------------------------------
I have another local working repo for my web server with a single master branch that i can push to two different remotes with two different file paths. This works as you would expect.
-------------------------------------------------------------------------------------------------------------------------------
SO :- 
1. Why is my local dev branch push not reflected on the file system of the remote but the master is?
2. Can I checkout different remote branches somehow?
            There is no working branch as the remote repo is bare.
             Can you locally switch the remote branch that changes the file system?
              i.e. get remote system files to reflect the local master or local dev branches
3. What is the correct/easiest way to push small changes to a remote file system without altering your local master branch?

Any help, clarification appreciated as I'm new to git.

**Comment by lbmeira** [comment]
This worked for me.
Try:
`git reset HEAD~1`

And then add/commit/push again

**Comment by moqiseh** [comment]
hi 
i had the same problem
you might change something in the prime without pushing it to hub
git pull (in prime )
git commit -a (in prime )
git push hub master (in prime )

then

git pull (in local)
git commit  -a (in local)
git push (in local)


it worked for me

**Comment by nitaria** [comment]
Sabtu, 04 Maret 2017, moqiseh <notifications@github.com> menulis:

**Comment by dr4dr4** [comment]
First thanks for reply, I have avoided the problem by not pushing a branch anymore.
The git reset head did not work for my case.
I tried ...
        git remote set-head <remote_name> <branch> 
worked in that it set a remote branch to head, but still did not behave as I would have expected.
I tried...
         git push --set-upstream <remote_name> <branch>
again seemed to work but did not work as I wanted.
The only thing that worked for what I wanted was below:-
         git symbolic-ref HEAD refs/heads/dev
Done on bare remote repo it then required a pull on local then it worked as I would have expected.

When I say work as I would have expected I mean I push from local branch to remote then login to remote and see my files there in the file system.
In the end I am opting for keeping it simple so when I push to master I know my file will be there where I pushed it in the file system.

To summarize 

I can make a branch on local
I can checkout this branch
I can push this to a remote server with same branch name
It appears that I can reset remote head etc.. but files don't appear on the remote machine
Only by doing the symbolic-ref statement in the remote/bare_repo/ dir can I get it to reflect the files in the file system.

SO I suppose what I am saying is...
Is there a way to set the remote repo so that it shows the files pushed by the local branch from the local machine?
For now I have two setups and avoid pushing branches to remotes in both.

**Comment by NateMS** [comment]
Did you try to setting the rights to execute?

**Comment by madhurivepa** [comment]
I had the same problem. The following worked:-
git reset HEAD~1
git pull
add/commit/push

**Comment by kboygit** [comment]
I had observed that recently, I think it is just a delay on the changes

**Comment by alperdemirci** [comment]
Hey @kboygit 
I am having the same problem, do you know how long is the delay?

**Comment by alperdemirci** [comment]
Nevermind, I figured it out. It took couple of min and I restarted the github. 
Now all my commits are in the repo.

**Comment by kboygit** [comment]
@alperdemirci glad you figured it out, it is around a minute and a half before you see the changes.

**Comment by wasimlovesruby** [comment]
i'm facing the same issue, anyone here, i need help, i have changed my code but after git push commit it didn't change, help?

**Comment by youhour** [comment]
The problem I had is that I do a push with "git push origin HEAD:refs/for/master".  Well this push will move the code to somewhere for code review.  

I ran below steps
git reset HEAD~1
add/commit/push  (push with "git push origin HEAD:refs/heads/maste")
Now I see my change in my github project.

**Comment by RichardBronosky** [comment]
How do @sinbad's last 2 comments have so many down votes? Does no one know how git works?

![Thats not how any of this works](https://i.imgur.com/CDqD1KV.jpg)

Clearly what @prashantakerkar was downloading was a `git clone` working tree of the git repo. So, what needed to happen is that the working tree needed a `git pull`. Depending on the backend in use, it may be possible to put a [post-receive](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks#__code_post_receive_code) server-side hook in place that would `cd` into the working tree and do a `git pull`.

**Comment by Pau1fitz** [comment]
the issue for me was that the incorrect email was linked to my account . To fix this I did the following:

`git config --global user.email "email@example.com"`

More instructions are [here](https://help.github.com/articles/setting-your-commit-email-address-in-git/).

**Comment by NettSite** [comment]
I have the same issue, this is what I have done:
`
In "origin" folder:

git config --global user.email "william@nettsite.co.za"
git init
git add .
git commit -m "First commit"
git config receive.denyCurrentBranch ignore

Change to folder which will contain the new repo

git clone ~/path-to-master/.git ~/path-to-target

Change to target folder

git add .
git commit -m "Commit Message"

Files from the origin are present in the new repo

Edit a file in the new repo

git commit -a -m "Commit Message"
git push origin master
Counting objects: 7, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (3/3), done.
Writing objects: 100% (7/7), 649 bytes | 0 bytes/s, done.
Total 7 (delta 1), reused 0 (delta 0)
To /home/william/git/live
   549ee0b..5398840  master -> master


Change to origin folder

git commit -a -m "Live Commit"
[master 9185c63] Live Commit
 1 file changed, 2 deletions(-)

Examine the file which was edited, the change is not reflected.
`
What am I doing wrong?

**Comment by ttaylorr** [comment]
Hi @NettSite, thanks for posting here. I think the issue in what's going on is here:

> Change to folder which will contain the new repo
>
> ``` 
> git clone ~/path-to-master/.git ~/path-to-target
> ```

By cloning from a local path _on disk_ (and not over the network) you'll get a remote named `origin` in your new copy of the repository that points to a path on disk. Git LFS does not support on-disk remotes, only networked ones. This is something that I plan on adding support for in the future, but I have not yet had the time.

**Comment by ttaylorr** [comment]
I'm going to close this issue for now, since I think that it has gotten somewhat off-topic from the original posting. Anyone who has participated in this thread is encouraged to open new issues if the problems persist, or you're interested in resurrecting it.
